lang/cs/Org.Apache.REEF.Common/TeleMetry/Counters.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    public class Counters : ICounters
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(Counters));
+
+        /// <summary>
+        /// It contains name and count pairs
+        /// </summary>
+        private readonly IDictionary<string, ICounter> _counters = new Dictionary<string, ICounter>();
+
+        /// <summary>
+        /// The lock for counters
+        /// </summary>
+        private readonly object _counterLock = new object();
+
+        [Inject]
+        private Counters()
+        {
+        }
+
+        public Counters(string serializedCountersString)
+        {
+            var c = JsonConvert.DeserializeObject<IEnumerable<Counter>>(serializedCountersString);
+            foreach (var ct in c)
+            {
+                _counters.Add(ct.Name, ct);
+            }
+        }
+
+        public IEnumerable<ICounter> GetCounters()
```

Comment:
The receiver side always gets ICounters. If this method is not defined in the interface, it cannot be called with the interface reference. Interface methods have to be public.